test/DynamicFees.t.sol

```diff
@@ -307,7 +301,22 @@ contract TestDynamicFees is Test, Deployers, GasSnapshot {
 
         BalanceDelta delta = swapRouter.swap(key, params, testSettings, ZERO_BYTES);
 
-        uint256 expectedProtocolFee = uint256(uint128(-delta.amount0())) * protocolFee0 / 1e6;
+        uint24 swapFee = uint16(protocolFee).calculateSwapFee(lpFee);
+
+        uint256 expectedProtocolFee;
+        int128 delta0 = -delta.amount0();
+
+        if (lpFee == 0) {
+            assertEq(protocolFee0, swapFee);
+            assembly {
```

Comment:
i think in the tests i'd feel happier doing this in solidity to check our assembly in the core code is correct/
something like

```
        expectedProtocolFee = uint256(uint128(delta0)) * protocolFee0 / 1e6;
        if (lpFee == 0) {
            assertEq(protocolFee0, swapFee);
            if (((delta0 * protocolFee0) % 1e6) != 0) expectedProtocolFee++
```